javalib/src/main/scala/java/util/IdentityHashMap.scala

```diff
@@ -0,0 +1,297 @@
+      val iter = iterator()
+      var changed = false
+      while (iter.hasNext) {
+        val elem = iter.next()
+        if (!findSame(elem, c)) {
+          iter.remove()
+          changed = true
+        }
+      }
+      changed
+    }
+
+    override def clear(): Unit =
+      self.clear()
+  }
+
+  private final class KeySet extends AbstractSet[K] {
+    def iterator(): Iterator[K] = {
+      new ju.Iterator[K] {
+        private val iter = inner.keyIterator()
+
+        def hasNext(): Boolean =
+          iter.hasNext()
+
+        def next(): K =
+          iter.next().inner
+
+        def remove(): Unit =
+          iter.remove()
+      }
+    }
+
+    def size(): Int =
+      self.size()
+
+    override def contains(o: Any): Boolean =
+      containsKey(o)
+
+    override def remove(o: Any): Boolean = {
+      val hasKey = contains(o)
+      if (hasKey) self.remove(o)
+      hasKey
+    }
+
+    override def removeAll(c: Collection[_]): Boolean =
+      if (size > c.size) {
+        c.scalaOps.foldLeft(false)((prev, elem) => this.remove(elem) || prev)
+      } else {
+        @tailrec
+        def removeAll(iter: Iterator[K], modified: Boolean): Boolean = {
+          if (iter.hasNext) {
+            if (findSame(iter.next(), c)) {
+              iter.remove()
+              removeAll(iter, true)
+            } else
+              removeAll(iter, modified)
+          } else
+            modified
+        }
+        removeAll(this.iterator, false)
+      }
+
+    override def retainAll(c: Collection[_]): Boolean = {
+      val iter = iterator()
+      var changed = false
+      while (iter.hasNext) {
+        val elem = iter.next()
+        if (!findSame(elem, c)) {
+          iter.remove()
+          changed = true
+        }
+      }
+      changed
+    }
+
+    override def clear(): Unit =
+      self.clear()
+  }
+
+  private final class EntrySet extends AbstractSet[Map.Entry[K, V]] {
+    def iterator(): Iterator[Map.Entry[K, V]] = {
+      new ju.Iterator[Map.Entry[K, V]] {
+        private val iter = inner.entrySetIterator()
+
+        def hasNext(): Boolean =
+          iter.hasNext()
+
+        def next(): Map.Entry[K, V] =
+          new MapEntry(iter.next())
+
+        def remove(): Unit =
+          iter.remove()
+      }
+    }
+
+    def size(): Int =
+      inner.size()
+
+    override def contains(value: Any): Boolean =
+      value match {
+        case value: Map.Entry[_, _] =>
+          val thatKey = value.getKey()
+          self.containsKey(thatKey) &&
+            same(self.get(thatKey), value.getValue())
+        case _ =>
+          false
+      }
+
+    override def remove(value: Any): Boolean =
+      value match {
+        case value: Map.Entry[_, _] =>
+          val thatKey = value.getKey()
+          val thatValue = value.getValue()
+          if (self.containsKey(thatKey) && same(self.get(thatKey), thatValue))
+              same(self.remove(thatKey), thatValue)
```

Comment:
I fixed that. The way I had it is someone in a multi-threaded way removed it between the `if` and `remove` then it would return `false` if the item was not null but the docs don't say that anyway so just removing and returning `true` make more sense.